feat(dipu,droplet): auto detecting PCCL library (#802)
DeepLink-org/deeplink.framework: dipu/torch_dipu/csrc_dipu/vendor/droplet/CMakeLists.txt

cmake_minimum_required(VERSION 3.5)

set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)

set(DROPLET_INSTALL "/usr/local/tangrt")
include(cmake/FindDropletCompiler.cmake)

find_package(PCCL)
if (PCCL_FOUND) 
    message(STATUS "use PCCL to implement communicator")
    message(STATUS "PCCL_LIBRARY: " ${PCCL_LIBRARY})
    message(STATUS "PCCL_INCLUDE_DIR: " ${PCCL_INCLUDE_DIR})
    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_PCCL" PARENT_SCOPE)
    set(DIPU_VENDOR_LIB ${DIPU_VENDOR_LIB} ${PCCL_LIBRARY})
    set(VENDOR_INCLUDE_DIRS ${VENDOR_INCLUDE_DIRS} ${PCCL_INCLUDE_DIR})
else()

>     message(STATUS "do not found PCCL to implement communicator, i.e. not support distributed on droplet")

Reviewer: ```suggestion
    message(STATUS "cannot found PCCL to implement communicator, i.e. not support distributed on droplet")
```

cmake_minimum_required(VERSION 3.5)

set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)

set(DROPLET_INSTALL "/usr/local/tangrt")
include(cmake/FindDropletCompiler.cmake)

> set(USE_PCCL "ON" CACHE STRING "Used to control whether to auto load and compile PCCL" FORCE)

Reviewer: 设置了应该是强制生效，而不是还要检查，不设置才是自动检测

find_package(PCCL)

> if (USE_PCCL == "ON" OR PCCL_FOUND) 

Reviewer: cmake string 比较不是这么写的，你写完得试试呀

选项应该是 on off auto，判断之前 tolower

如果 on 了，那么应该是 required

Reviewer: > 所以为啥不直接用 bool

为了 AUTO?

> if (USE_PCCL == "ON" && PCCL_FOUND) 

Reviewer: 语法不对吧

cmake_minimum_required(VERSION 3.5)

set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)

set(DROPLET_INSTALL "/usr/local/tangrt")
include(cmake/FindDropletCompiler.cmake)

set(USE_PCCL "AUTO" CACHE STRING "Used to force load and compile PCCL. AUTO: auto load and compile PCCL; 1(TRUE or true): Force load and compile PCCL; 0(FALSE or false): Forbidden load PCCL")

find_package(PCCL)

> if (USE_PCCL OR USE_PCCL STREQUAL "AUTO")

Reviewer: ```suggestion
if (USE_PCCL)
```

>     if (USE_PCCL AND  NOT PCCL_FOUND)
>         message(STATUS "Cannot find PCCL to implement communicator, not support distributed on droplet!")
>     endif()

Reviewer: 注意带不带引号的不同
```suggestion
    if ("${USE_PCCL}" AND NOT PCCL_FOUND)
        message(FATAL_ERROR "Set to force compiling with PCCL, but PCCL is not found.")
    endif()
```

    if(PCCL_FOUND)
        message(STATUS "Use PCCL to implement communicator")
        message(STATUS "PCCL_LIBRARY: " ${PCCL_LIBRARY})
        message(STATUS "PCCL_INCLUDE_DIR: " ${PCCL_INCLUDE_DIR})
        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_PCCL" PARENT_SCOPE)
        set(DIPU_VENDOR_LIB ${DIPU_VENDOR_LIB} ${PCCL_LIBRARY})
        set(VENDOR_INCLUDE_DIRS ${VENDOR_INCLUDE_DIRS} ${PCCL_INCLUDE_DIR})

>     elseif(USE_PCCL STREQUAL "AUTO")
>         message(STATUS "Cannot auto load PCCL to implement communicator, not support distributed on droplet!")
>     endif()

Reviewer: 这里简化吧，不判断 auto 了，错误输入 fallback 到 auto 也合理
```suggestion
    else()
        message(STATUS "PCCL not found, i.e. not supporting distributed on droplet")
    endif()
```

else()

>     message(STATUS "Cannot find PCCL to implement communicator, i.e. disabled by env vars(USE_PCCL)!")

Reviewer: ```suggestion
    message(STATUS "Force compiling without PCCL, i.e. not supporting distributed on droplet")
```
